packages/vite/src/node/preview.ts

```diff
@@ -127,6 +129,11 @@ export async function preview(
   app.use(previewBase, async (req, res, next) => {
     if (shouldServe(req.url!, distDir)) {
       return assetServer(req, res, next)
+    } else if (spaFallback) {
+      const indexHtml = fs.readFileSync(resolve(distDir, 'index.html'), 'utf-8')
+      res.write(indexHtml)
+      res.end()
+      return
```

Comment:
I think we'd want to wrap a try-catch in case there's no `index.html`, and to preserve the previous behaviour too.

Also would be great to use `path.resolve` so we don't need a new named import.

```suggestion
    } else if (spaFallback) {
      try {
        const indexHtml = fs.readFileSync(path.resolve(distDir, 'index.html'), 'utf-8')
        res.write(indexHtml)
        res.end()
      } catch {}
      return
```